test/Interop/Cxx/constructor/copy-constructor-irgen.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: %target-swift-frontend -enable-cxx-interop -I %S/Inputs %s -emit-ir | %FileCheck %s
+
+import Basic
+
+// CHECK-LABEL: define swiftcc void @"$s4main4testSo30HasUserProvidedCopyConstructorVyF"
+// CHECK: [[OBJ:%.*]] = alloca %TSo30HasUserProvidedCopyConstructorV
+// Make sure we bitcast it from the Swift type to the C++ type.
+// CHECK: [[CXX_ARG:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* %0 to %struct.HasUserProvidedCopyConstructor*
+// CHECK: [[CXX_OBJ:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* [[OBJ]] to %struct.HasUserProvidedCopyConstructor*
+// Make sure we call the destructor.
+// CHECK: call void @_ZN30HasUserProvidedCopyConstructorC1ERKS_(%struct.HasUserProvidedCopyConstructor* [[CXX_ARG]], %struct.HasUserProvidedCopyConstructor* [[CXX_OBJ]])
+// CHECK: ret void
+public func test() -> HasUserProvidedCopyConstructor {
+  let obj = HasUserProvidedCopyConstructor(42)
+  return obj
```

Comment:
I'm not sure I understand why this code must call the copy constructor. Swift's frontend could (and ideally should) elide the extra copy. I'd suggest something like this, to ensure that a copy is semantically needed, and to avoid constructing the object in the test (which generates code that is not relevant to the test):

```
public func test(input: HasUserProvidedCopyConstructor) -> (HasUserProvidedCopyConstructor, HasUserProvidedCopyConstructor) {
  return (input, input)
}
```

Reply:
I like this a lot better. Implemented.